### PR TITLE
feat: Add icons for executables.

### PR DESCRIPTION
**Pull request by DreepyYunky**
This adds icons for executable files.

This includes:

- `.exe` for Windows Executables.
- `.app` for MacOS.
- `.out` and `.elf` for Linux.

Non-related change: I changed the GDScript and Godot project files to it's NerdFont godot icon.

This is my first contribution, and I just started studying the code today, so please tell me if I did anything wrong.

### CODE REVIEW

**Comment by gegoune** [comment]
Thanks! Would you mind breaking non-exec related change into two separate PRs?

**Comment by DreepyYunky** [comment]
> Thanks! Would you mind breaking non-exec related change into two separate PRs?

Ok, no problem. I'm going to revert the Godot-related changes.

File: lua/nvim-web-devicons/icons-light.lua [hunk 1]
```diff
@@ -380,6 +380,13 @@ local icons_by_file_extension = {
     cterm_color = "58",
     name = "Ai",
   },
+  ["app"] = {
+    icon = "",
+    color = "#9F0500",
+    cterm_color = "196",
+    name = "App",
+  },
+
```
Comment (gegoune) [inline lua/nvim-web-devicons/icons-light.lua:389]:
```suggestion
```

File: lua/nvim-web-devicons/icons-light.lua [hunk 2]
```diff
@@ -686,12 +693,20 @@ local icons_by_file_extension = {
     cterm_color = "58",
     name = "Ejs",
   },
+  ["elf"] = {
+    icon = "",
+    color = "#9F0500",
+    cterm_color = "196",
+    name = "Elf",
+  },
+
```
Comment (gegoune) [inline lua/nvim-web-devicons/icons-light.lua:702]:
```suggestion
```

File: lua/nvim-web-devicons/icons-light.lua [hunk 3]
```diff
@@ -686,12 +693,20 @@ local icons_by_file_extension = {
...
   ["elm"] = {
     icon = "",
     color = "#36677c",
     cterm_color = "24",
     name = "Elm",
   },
+
```
Comment (gegoune) [inline lua/nvim-web-devicons/icons-light.lua:709]:
```suggestion
```

File: lua/nvim-web-devicons/icons-light.lua [hunk 4]
```diff
@@ -722,6 +737,13 @@ local icons_by_file_extension = {
     cterm_color = "96",
     name = "Ex",
   },
+  ["exe"] = {
+    icon = "",
+    color = "#9F0500",
+    cterm_color = "196",
+    name = "Exe",
+  },
+
```
Comment (gegoune) [inline lua/nvim-web-devicons/icons-light.lua:746]:
```suggestion
```

File: lua/nvim-web-devicons/icons-light.lua [hunk 5]
```diff
@@ -1226,6 +1248,13 @@ local icons_by_file_extension = {
     cterm_color = "236",
     name = "OpenTypeFont",
   },
+  ["out"] = {
+    icon = "",
+    color = "#9F0500",
+    cterm_color = "196",
+    name = "Out",
+  },
+
```
Comment (gegoune) [inline lua/nvim-web-devicons/icons-light.lua:1257]:
```suggestion
```

**Review by gegoune — CHANGES_REQUESTED** [review]

File: lua/nvim-web-devicons/icons-light.lua (hunk 5 above)
Comment (DreepyYunky) [reply lua/nvim-web-devicons/icons-light.lua:1257]: Working on it.

Comment (DreepyYunky) [reply lua/nvim-web-devicons/icons-light.lua:1253]: There. Every space should've been removed.

File: lua/nvim-web-devicons/icons-default.lua [hunk 6]
```diff
@@ -717,11 +729,17 @@ local icons_by_file_extension = {
     name = "Erl",
   },
   ["ex"] = {
-    icon = "",
```
Comment (gegoune) [inline lua/nvim-web-devicons/icons-default.lua:720]: Indentation.

**Review by gegoune — CHANGES_REQUESTED** [review]

**Comment by alex-courtis** [comment]
Looks like the light colours haven't been properly generated: https://github.com/nvim-tree/nvim-web-devicons/actions/runs/7220267899/job/19672971183?pr=351

Please run `make` or `make colors` to generate the cterm and light colours and commit the updates. See https://github.com/nvim-tree/nvim-web-devicons/blob/master/CONTRIBUTING.md#building

**Comment by alex-courtis** [comment]
I'm not actually seeing those icons with HackNerdFontMono-Regular.ttf

Which font are you using? Are you sure they are glyphs from nerd fonts?

**Comment by gegoune** [comment]
Same for me on Iosevka.

**Comment by DreepyYunky** [comment]
> I'm not actually seeing those icons with HackNerdFontMono-Regular.ttf
> 
> Which font are you using? Are you sure they are glyphs from nerd fonts?
> 
> 

I use Gohu uni14 NerdFont

**Comment by gegoune** [comment]
Thank you!

**Comment by pysan3** [comment]
Hi, I get this error now.

```text
...re/nvim/lazy/nvim-web-devicons/lua/nvim-web-devicons.lua:253: Invalid highlight color: '#f28ba28'
```

What's up with the 7th digit?

**Comment by gegoune** [comment]
Followed up in https://github.com/nvim-tree/nvim-web-devicons/pull/357.

**Comment by alex-courtis** [comment]
> Followed up in #357.
> 
> @alex-courtis Need to make sure ci picks those up in the future.

#359

**Pull request merged**